src/commandline_options.cpp

```diff
@@ -595,18 +598,18 @@ void commandline_options::parse_resolution_ ( const std::string& resolution_stri
 		throw bad_commandline_resolution(resolution_string);
 	}
 
-	resolution = std::pair(xres, yres);
+	resolution.emplace(xres, yres);
 }
 
 std::vector<std::pair<unsigned int,std::string>> commandline_options::parse_to_uint_string_tuples_(const std::vector<std::string> &strings, char separator)
 {
 	std::vector<std::pair<unsigned int,std::string>> vec;
-	const std::string& expected_format
+	const std::string expected_format
```

Comment:
This would also be a good place to use string literals.

```diff
-			= std::string() + "UINT" + separator + "STRING";
+			= "UINT"s + separator + "STRING"s;
```

(Requires a `using namespace std::literals` to be in scope.)

Reply:
on it

Reply:
done

Reply:
@CelticMinstrel does it subtly matter whether the `"STRING"` is marked as a literal instead of a C-style string?

Looks okay as-is to me, I assume the compiler will optimise anyway.

Reply:
I don't think it makes any actual difference to the resulting code. If you omit the second `s`, the compiler will still generate an implicit conversion to string there, I think, so the `s` just makes the conversion explicit. In any case, even if I'm wrong, I'm pretty sure it would come out the same after optimization.